xpresso/binders/_body/extractors/discriminated.py

```diff
@@ -53,11 +53,15 @@ def register_parameter(self, param: inspect.Parameter) -> BodyExtractor:
                 annotation=arg,
                 default=param.default,
             )
-            marker: typing.Optional[BodyBinderMarker] = None
-            for param_marker in get_markers_from_parameter(sub_body_param):
-                if isinstance(param_marker, BodyBinderMarker):
-                    marker = param_marker
-                    break
+            marker: typing.Optional[BodyBinderMarker] = next(
+                (
+                    param_marker
+                    for param_marker in get_markers_from_parameter(sub_body_param)
+                    if isinstance(param_marker, BodyBinderMarker)
+                ),
+                None,
+            )
```

Comment:
Same as the comment above: I prefer the more verbose language neutral for loop with early exit v.s. the language specific use of `next(..., default=None)`, allocating a generator, etc.